### PR TITLE
Removed ReactiveList mention from documentation.

### LINKED ISSUE #159: Make a Work in Progress PR to remove ReactiveList
ReactiveList<T> is going away shortly.

We should have a document page pointing towards Roland Pheasants DynamicData page as a alternative https://github.com/RolandPheasant/DynamicData/wiki/Introduction-for-ReactiveUI-users and also https://github.com/RolandPheasant/DynamicData

Also we should remove ReactiveList, IReactiveCollection, IReactiveDeriveList etc all from our documentation. 

As alternatives we should be free to use DynamicData as example etc (it's likely it will become part of the ReactiveUI family anyway soon).

### PR DESCRIPTION
**Pull request by glennawatson**
Fixes #159

Removes mention of the ReactiveList from the documentation.

Updated one of the samples to actually work (was broken) and use DynamicData.

### CODE REVIEW

File: input/docs/handbook/message-bus/index.md [hunk 1]
```diff
@@ -83,19 +92,33 @@ public class DocumentViewModel : ReactiveObject
...
     {
-        OpenDocuments = new ReactiveList<DocumentViewModel>();
+        OpenDocuments = new ObservableCollection<DocumentViewModel>();
 
         // Whenever the list of documents change, calculate a new Observable
         // to represent whenever any of the *current* documents have been
         // requested to close, then Switch to that. When we get something
         // to close, remove it from the list.
-        OpenDocuments.Changed
+        OpenDocuments.ToObservableChangeSet()
```
Comment (RLittlesII) [inline input/docs/handbook/message-bus/index.md]: drop the `ToObservableChangeSet()` down a line for readability.

File: input/docs/handbook/collections/derived-lists.md [hunk 2]
```diff
@@ -1,98 +0,0 @@
-# CreateDerivedCollection and derived lists
```
[the remark this answers is not in the document]
Comment (glennawatson) [reply input/docs/handbook/collections/derived-lists.md]: I'm happy to do redirects, but I don't think we have the time and resources to cater for older versions of documentation etc.

**Comment by worldbeater** [comment]
Maybe we shouldn't completely remove `ReactiveList` docs, maybe we should leave a page for it for those users who have to support projects with legacy codebase. You may disagree, but marking `ReactiveList` page as `[Obsolete]` plus rewriting code examples to use `DynamicData` plus bringing  DynamicData docs to the website may be just fine I think

**Comment by glennawatson** [comment]
I added a Obsolete folder where we can put pages that are being deprecated. 

I brought over the DD RxUI guide and modified it like I said.

File: input/docs/handbook/collections/index.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+ReactiveUI recommends the use of DynamicData framework for collection based operations. DynamicData has replaced internally the use of [ReactiveList](../obsolete/collections)
+
+# Overview of Dynamic Data
```
[the remark this answers is not in the document]
Comment (RLittlesII) [reply input/docs/handbook/collections/index.md]: @RolandPheasant Added your feedback

**Pull request merged**